lib/src/main/java/com/exercism/runner/TestRunner.java

```diff
@@ -0,0 +1,67 @@
+package com.exercism.runner;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import com.exercism.data.Report;
+import com.exercism.junit.JUnitTestParser;
+import com.exercism.report.ReportGenerator;
+import com.exercism.xml.JUnitXmlParser;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import com.google.common.io.MoreFiles;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public final class TestRunner {
+    private static final String GRADLE_TEST_ERR = "gradle-test.err";
+
+    public static void main(String[] args) throws InterruptedException, IOException {
+        run(args[0]);
+    }
+
+    private static void run(String slug) throws InterruptedException, IOException {
+        Process gradleTest = new ProcessBuilder(
+            "gradle",
+            "test",
+            "--offline",
+            "--no-daemon",
+            "--warning-mode=none")
+            .redirectError(new File(GRADLE_TEST_ERR))
+            .start();
+        if (!gradleTest.waitFor(20, SECONDS)) {
+            throw new IllegalStateException("gradle test did not complete within 20 seconds");
```

Comment:
I had to change this to 20 seconds on my machine for it so pass.

Reply:
That is mildly concerning since the whole point is that this part needs to take a shorter amount of time than it was taking before. Perhaps there are even more options that need to be passed to "gradle test" that would make it go faster?

Reply:
I first want to check how fast it runs on our Exercism infrastructure. It could be my local setup that is slow.